### PR TITLE
usbus: Add USB peripheral serial string support

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

This adds compile-time options to configure the serial of an USB
peripheral. The serial be autogenerated with a configured number of
bytes. It is also possible to configure a fixed serial string for a
device and disable the autogeneration of the serial.

I've added Kconfig support for the new options. @leandrolanzieri: Could you verify that what I have created is the right way if I want `CONFIG_USB_SERIAL_STR` only emitted if it has a non-zero value.

### Testing procedure

usbus_minimal and some `lsusb -v -d 1209:7d01` can be used to verify the reported serial.

### Issues/PRs references

None

### CODE REVIEW

File: sys/usb/Kconfig [hunk 1]
```diff
@@ -81,6 +81,31 @@ config USB_DEFAULT_LANGID
...
+        devices with identical VID:PID combination.
+
+config USB_SERIAL_STR
+    string
+    default USB_SERIAL_STR_KCONFIG_ENTRY
+    depends on USB_SERIAL_STR_KCONFIG_ENTRY != ""
+
+config USB_SERIAL_BYTE_LENGTH
+    int "USB peripheral serial string length"
+    range 0 128
```
Comment (dylad) [inline sys/usb/Kconfig:103]: In theory, we could set it between 0 and 255.

**Review by dylad — COMMENTED** [review]
USB part looks good. Work as expected.
I'll leave the Kconfig review to our experts.

**Comment by bergzand** [comment]
The maximum number of bytes to use for the string is 63, USB descriptor strings are UTF16LE encoded, every byte from the luid becomes 4 bytes of string descriptor. With a maximum of 255 bytes for a (non-null-terminated) string, this means that we can use at most 63 bytes for the unique string.

File: sys/usb/Kconfig [hunk 2]
```diff
@@ -81,6 +81,31 @@ config USB_DEFAULT_LANGID
     help
         The default value is EN-US (0x0409).
 
+config USB_SERIAL_STR_KCONFIG_ENTRY
```
Comment (leandrolanzieri) [inline sys/usb/Kconfig:84]: Would something like the following work instead?
```
config USB_CUSTOM_SERIAL_STR
    bool "Configure a custom serial string"

config USB_SERIAL_STR
    string "Serial string"
    depends on USB_CUSTOM_SERIAL_STR
    help
        USB peripheral serial string

        Compile-time value to override the serial string with. An LUID-based hex
        string is generated when this value is not used.

        This string does not have to be a number, but it must be unique between
        devices with identical VID:PID combination.

config USB_SERIAL_BYTE_LENGTH
    int "USB peripheral serial string length"
    depends on !USB_CUSTOM_SERIAL_STR
    range 0 63
    default 8
    help
        The length here is in bytes. The generated hex string is twice as many chars
        in length due to conversion from bytes to hex chars.

comment "WARNING: The serial string is empty!"
    depends on USB_SERIAL_STR = "" && USB_CUSTOM_SERIAL_STR
```

Comment (bergzand) [reply sys/usb/Kconfig:84]: That would make the decision between fixed string and generated hex string more explicit, I like it.

Comment (bergzand) [reply sys/usb/Kconfig]: Adapted to this. I'm happy with the current menu structure.

**Comment by leandrolanzieri** [comment]
Kconfig part looks good to me, feel free to squash @bergzand

**Comment by bergzand** [comment]
Squashed!

**Review by dylad — APPROVED** [review]
ACK.
Tested work on same54-xpro (random luid serial and custom string from Kconfig), the USB part looks good to me.

**Pull request merged**